ui/src/message_popup/popup_question.jsx

```diff
@@ -49,221 +47,88 @@ export default React.createClass({
-        <div style={styles.question}>{text}</div>
-        {scaffolding.shouldShowStudentCard && student &&
-          <StudentCard useCardStyles={true} student={student} />}
-        {scaffolding.helpType === 'hints' && 
-          <div style={styles.hintCard}>
-            <HintCard examples={examples} nonExamples={nonExamples} />
-          </div>}
-        <div style={styles.textAreaContainer}>
-          <TextField
-            style={styles.textField}
-            textareaStyle={styles.textareaInner}
-            underlineShow={false}
-            floatingLabelText='Speak directly to the student'
-            onChange={this.onTextChanged}
-            multiLine={true}
-            disabled={this.state.isRevising} 
-            rows={2}/>
-        </div>
-        <div style={styles.buttonRow}>
-          <RaisedButton
-            onTouchTap={this.onSavePressed}
-            style={styles.button}
-            secondary={true}
-            label={scaffolding.helpType === 'feedback' ? 'Save Response' : 'Respond'}
-            disabled={this.state.isRevising || this.state.initialResponseText === ''}/>
-          {seconds > 0 &&
-            <div style={styles.ticker}>{seconds}s</div>
-          }
-        </div>
-        <VelocityTransitionGroup enter={{animation: "slideDown"}} runOnMount={true}>
-          {this.state.isRevising &&
-            <div style={styles.feedbackCard}>
-              <FeedbackCard
-                initialResponseText={this.state.initialResponseText}
-                onRevised={this.onRevised}
-                onPassed={this.onPassed}
-                examples={examples}/>  
-            </div>}
-        </VelocityTransitionGroup>
-        <Snackbar
-          open={seconds >= (this.props.limitMs/1000)}
-          message="Remember, these are supposed to be quick responses."
-          onRequestClose={this.onRequestClose}
-          />
+        <ScenarioRenderer
+          question={this.props.question} />
+        <PromptsRenderer
+          scaffolding={scaffolding}
+          student={student}
+          question={this.props.question}
+        />
+        <ResponseRenderer
+          scaffolding={scaffolding}
+          question={question}
+          limitMs={limitMs}
+          onResponseSubmitted={this.onResponseSubmitted}
+          onLogMessage={this.logData}
+        />
       </div>
     );
   },
   
-  renderSummary(){
+  renderSummary() {
+    const {elapsedMs, finalResponseText, initialResponseText} = this.state.response;
+    const {question, scaffolding, isLastQuestion} = this.props;
+
     return (
       <SummaryCard 
-        onDone={this.onDonePressed} 
-        question={this.props.question} 
-        response={this.props.scaffolding.helpType==='feedback' ? this.state.finalResponseText : this.state.initialResponseText} 
-        elapsedSeconds={Math.round(this.state.elapsedMs / 1000)} 
-        buttonLabel={this.props.isLastQuestion ? "Finish" : "Next Question"}/>
+        onDone={this.onDone} 
```

Comment:
Good catch!  I ran into something similar adding in video scenarios, and so I'll make the minimal change to fix this and then I think we'll have to rework this a bit.

This is a place where tests can help too, when there are lots of different combinations and we want to check that we don't break one when reworking another.